controllers/adminCommandersController.js

```diff
@@ -54,46 +56,32 @@ exports.index = function(req, res) {
 
 //Updates a users permission level.
 exports.updateUser = function(req, res) {
-    if (req.isAuthenticated() && req.user.roleNumeric > 4) {
-        esi.characters.search.strict(req.body.pilotName).then(function (results) {
-            users.updateUserPermission(results[0], req.body.permission, req.user, res)
-            {
-                req.flash("content", {"class":"success", "title":"User permission updated.", "message":"Tell the user to refresh their browser twice for the changes to take effect."});
-                res.redirect('/admin/commanders');
-            }
-        }).catch(function (err) {
-            log.error("routes.post: Error for esi.characters.search", { err, name: req.body.name });
-            req.flash("content", {"class":"error", "title":"Woops!", "message":"We couldn't find " + req.body.pilotName + ". Please make sure they have logged in at least once before."});
-            res.status(409).redirect('/admin/commanders');
-        })
-    } else {
+    if(req.isAuthenticated() && !req.user.role.numeric >= 4)
+    {
         req.flash("content", {"class":"error", "title":"Not Authorised!", "message":"You are not allowed to adjust the permissions of this user. Think this is an error? Contact a member of leadership."});
         res.status(403).redirect('/admin/commanders');
+        return;
+    }
+    //Only allow senior FC to make people trainees
+    if(req.user.role.numeric == 4){
```

Comment:
I don't want to enum the `role.numeric` checks. It would only make sense to use the title names but they are set in the setup.js file and they are not constant. 

I would sooner make some helper methods `isFC()` (role > 0). `isLeadership()` (role > 3) as well as the standard `isAuthenticated()` (used for any pilot)
```
//roleNumeric titles, use null to disable. Leave index0 as member
exports.userPermissions =  ["Member", "Trainee", null, "Fleet Commander", "Senior FC", "Leadership"];
```

Reply:
This has been resolved by adding a helper function
`users.isRoleNumeric(req.user, int)`. This will check to ensure the visitor is a user (replacing `req.isAuthenticated()` and then check that their value is greater than or equal to `<=` the int.